### PR TITLE
Change from jpa to jdbc and fix code accordingly

### PR DESCRIPTION
**Pull request by huiningd**
* Update libs: add jdbc, validation, remov jpa
* Use Kotlin data class
* Fix deck-collection endpoints handling from jpa to jdbc
* Change package structure from by layer to by feature
* Rename DeckCollection to DeckGroup

### CODE REVIEW

File: src/main/kotlin/com/example/flashcardbackend/deckgroup/DeckGroupRepository.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            resultSet.getString("description"),
+        )
+    }
+}
+
+@Transactional
+@Repository
+class DeckGroupRepository(@Autowired val jdbcTemplate: JdbcTemplate) {
+
+    fun findAll(): List<DeckGroupListItemDTO> {
```
Comment (petrikainulainen-nitor) [inline src/main/kotlin/com/example/flashcardbackend/deckgroup/DeckGroupRepository.kt]: This should return list of domain objects (maybe DeckGroupListItem?). This way your repository API stays consistent (domain object in, domain object out).

File: src/main/kotlin/com/example/flashcardbackend/deckgroup/DeckGroupService.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.example.flashcardbackend.deckgroup
+
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+class DeckGroupService(val repository: DeckGroupRepository) {
+
+    @Transactional(readOnly = true)
+    fun findDeckGroups(): List<DeckGroupListItemDTO> = repository.findAll()
```
Comment (petrikainulainen-nitor) [inline src/main/kotlin/com/example/flashcardbackend/deckgroup/DeckGroupService.kt]: The implementation of this service method should map list of domain objects into list of DeckGroupListItemDTO objects.

**Review by petrikainulainen-nitor — APPROVED** [review]
After you changed the one repository method that returns list of DTOs, feel free to merge this PR.

File: src/main/kotlin/com/example/flashcardbackend/deckgroup/DeckGroupRepository.kt (hunk 1 above)
Comment (huiningd) [reply src/main/kotlin/com/example/flashcardbackend/deckgroup/DeckGroupRepository.kt]: thanks for pointing it out! yeah, forgot to take care of the way out (domain object out). should have checked the code again before PR.

File: src/main/kotlin/com/example/flashcardbackend/deckgroup/DeckGroupService.kt (hunk 2 above)
Comment (huiningd) [reply src/main/kotlin/com/example/flashcardbackend/deckgroup/DeckGroupService.kt]: thanks for pointing it out! will fix :)

**Pull request merged**